### PR TITLE
[Rel-4_0 - Bug 10534] Wildcard hacking the customer information center

### PR DESCRIPTION
**Pull request by s7design**
http://bugs.otrs.org/show_bug.cgi?id=10534

Here is fixed problem with wiledcard ( eg 'asterisk', customer'asterisk', customer% ...) in CIC.
It was fixed in the old PR ( https://github.com/OTRS/otrs/pull/339 ), I think it is better to solve it in separate one.

Regards
Zoran

### CODE REVIEW

File: Kernel/System/CustomerUser/DB.pm [hunk 1]
```diff
@@ -299,6 +304,18 @@ sub CustomerSearch {
             $SQL .= "LOWER($Self->{CustomerID}) LIKE LOWER(?) $LikeEscapeString";
         }
     }
+    elsif ( $Param{CustomerIDRaw} ) {
+
+        my $CustomerIDRaw = $Self->{DBObject}->Quote( $Param{CustomerIDRaw}, 'Like' );
+        push @Bind, \$CustomerIDRaw;
+
+        if ( $Self->{CaseSensitive} ) {
+            $SQL .= "$Self->{CustomerID} = ? $LikeEscapeString";
```
Comment (mgruner) [inline Kernel/System/CustomerUser/DB.pm]: You are (correctly) making an exact match (=) here. Then why do you quote with 'Like' and pass the LikeEscapeString?

**Comment by s7design** [comment]
Hi @mgruner 

I fixed it, it was my mistake.

Regards
Zoran

**Comment by mgruner** [comment]
This looks nice. Please extend CustomerUser.t to also cover the new mode.

**Comment by s7design** [comment]
Hi @mgruner 

I extended the unit test for CustomerUser, I hope it is good. If you have any suggestion, how I can improve it, please let me know.

Regards
Zoran

**Comment by mgruner** [comment]
Merged, thank you!

**Pull request closed without merging**